src/rust/engine/options/src/arg_splitter.rs

```diff
@@ -15,8 +17,36 @@ pub const UNKNOWN_GOAL_NAME: &str = "__unknown_goal";
 
 lazy_static! {
     static ref SPEC_RE: Regex = Regex::new(r"[/\\.:*#]").unwrap();
-    static ref SINGLE_DASH_FLAGS: HashSet<&'static str> =
-        HashSet::from(["-ltrace", "-ldebug", "-linfo", "-lwarn", "-lerror", "-h", "-v", "-V"]);
+    static ref SINGLE_DASH_FLAGS: HashSet<&'static str> = HashSet::from([
+        "-h", "-v", "-V", "-ltrace", "-ldebug", "-linfo", "-lwarn", "-lerror", "-l=trace",
+        "-l=debug", "-l=info", "-l=warn", "-l=error",
+    ]);
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Args {
+    pub(crate) arg_strings: Vec<String>,
+}
+
+impl Args {
+    // Create an Args instance with the provided args, which must *not* include the
+    // argv[0] process name.
+    pub fn new<I: IntoIterator<Item = String>>(arg_strs: I) -> Self {
+        Self {
+            arg_strings: arg_strs.into_iter().collect(),
+        }
+    }
+
+    pub fn argv() -> Self {
+        let mut args = env::args().collect::<Vec<_>>().into_iter();
+        args.next(); // Consume the process name (argv[0]).
+                     // TODO: In Pants's own integration tests we may invoke Pants in a subprocess via
+                     //  `python -m pants` or `python path/to/__main__.py` or similar. So
+                     //  skipping argv[0] may not be sufficient to get just the set of args to split.
+                     //  In practice our tests pass despite these extra args being interpreted as specs
+                     //  or goals, but that is skating on thin ice.
+        Self::new(env::args().collect::<Vec<_>>())
```

Comment:
There may be a latent typo here, because it looks like the `args` variable is prepped but unused. I think either:

- `args` above is unnecessary and can be dropped
- `args` should be used here

```suggestion
        Self::new(args)
```

(Looks like this was inherited from the previous implementation.)

Reply:
D'oh!